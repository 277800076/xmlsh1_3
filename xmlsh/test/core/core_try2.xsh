# try/catch/finally test 2
# Test throw within a script

if [ $# -gt 0 ] ; then
	echo In sub-script 
	throw $1
	echo Fail SNH - after throw
fi

try {
	$0 test
} catch E {
	echo Successs got $E
} finally {
	echo Success in finally
}


func_throws () 
{
	echo in fun_throws
	throw $1
	echo Fail - SNH after throw
}


try {
	echo calling throw function
	func_throws <[ <foo/> ]>
	echo Fail - SNH after throw
} catch X
{
	echo Success caught $X
}



