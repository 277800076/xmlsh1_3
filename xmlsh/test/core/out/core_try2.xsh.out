In sub-script
Successs got test
Success in finally
calling throw function
in fun_throws
Success caught <foo/>
